src/lib.rs

```diff
@@ -1023,6 +1023,34 @@ mod test {
     }
 
     #[test]
+    fn self_signed_cert() {
+        let creds = SchannelCredBuilder::new()
+            .acquire(Direction::Outbound)
+            .unwrap();
+        let stream = TcpStream::connect("self-signed.badssl.com:443").unwrap();
+        let err = TlsStreamBuilder::new()
+            .domain("self-signed.badssl.com")
+            .initialize(creds, stream)
+            .err()
+            .unwrap();
+        assert_eq!(err.raw_os_error().unwrap(), winapi::CERT_E_UNTRUSTEDROOT as i32);
+    }
+
+    #[test]
+    fn wrong_host_cert() {
```

Comment:
There's already one of these up above at bad_domain - maybe get rid of that one?